web/src/app/pages/main-page-container/main-page/secondary-side-panel/submission-panel/submission-panel.component.html

```diff
@@ -34,59 +34,61 @@
-                <mat-checkbox disabled="true" checked="true" *ngFor="let option of getTaskMultipleChoiceSelections(task).values">
-                  {{option.label}}
-                </mat-checkbox>
-  
-                <mat-checkbox disabled="true" checked="true" *ngIf="getTaskMultipleChoiceSelections(task).otherValue as otherValue">
-                  Other: {{otherValue}}
-                </mat-checkbox>
-              </div>
-              <!-- Photo task types -->
-              <div *ngIf="task.type === taskType.PHOTO">
-                <img src="{{this.firebaseURLs.get(getTaskSubmissionResult(task)!.value)}}" class="submission-img"/>
+        <ng-container *ngIf="getTaskSubmissionResult(task) !== undefined">
+          <mat-divider *ngIf="i > 0"></mat-divider>
+          <div class="submission-task">
+            <!-- Question -->
+            <div class="submission-question">{{task.label}}</div>
+            <!-- Response -->
+            <div class="submission-response">
+              <div *ngIf="getTaskSubmissionResult(task).value === null" class="submission-no-response">
+                Skipped
               </div>
-              <!-- Geometry task types -->
-              <div *ngIf="task.type === taskType.DROP_PIN || task.type === taskType.CAPTURE_LOCATION || task.type === taskType.DRAW_AREA">
-                <div class="submission-response-geometry" [ngClass]="{'selected-submission-response-geometry': task.id === selectedTaskId}">
-                  <div class="submission-response-geometry-number" [ngStyle]="{ 'background-color': this.submission?.job.color ?? 'black' }">
-                    {{i}}
-                  </div>
-                  <div *ngIf="task.type === taskType.DROP_PIN || task.type === taskType.DRAW_AREA" class="submission-response-geometry-label">
-                    {{task.id}}
-                  </div>
-                  <div *ngIf="task.type === taskType.CAPTURE_LOCATION" class="submission-response-geometry-label">
-                    {{getCaptureLocationCoord(task)}}
+              <div *ngIf="getTaskSubmissionResult(task).value !== null">
+                <!-- Multiple choice task types -->
+                <div *ngIf="task.type === taskType.MULTIPLE_CHOICE" class="submission-response-multiple-choice">
+                  <mat-checkbox disabled="true" checked="true" *ngFor="let option of getTaskMultipleChoiceSelections(task).values">
+                    {{option.label}}
+                  </mat-checkbox>
+    
+                  <mat-checkbox disabled="true" checked="true" *ngIf="getTaskMultipleChoiceSelections(task).otherValue as otherValue">
+                    Other: {{otherValue}}
+                  </mat-checkbox>
+                </div>
+                <!-- Photo task types -->
+                <div *ngIf="task.type === taskType.PHOTO">
+                  <img src="{{this.firebaseURLs.get(getTaskSubmissionResult(task)!.value)}}" class="submission-img"/>
```

Comment:
We're exposing a lot of logic in the view here - how the URLs are stored in the component, how we get the result for a task, how we extract the value.. can we move this into the component in a follow-up PR?

Reply:
ok